src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -10972,8 +10976,10 @@ public abstract partial class TextReader : System.MarshalByRefObject, System.IDi
         public virtual System.Threading.Tasks.ValueTask<int> ReadBlockAsync(System.Memory<char> buffer, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual string? ReadLine() { throw null; }
         public virtual System.Threading.Tasks.Task<string?> ReadLineAsync() { throw null; }
+        public virtual System.Threading.Tasks.ValueTask<string?> ReadLineAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
         public virtual string ReadToEnd() { throw null; }
         public virtual System.Threading.Tasks.Task<string> ReadToEndAsync() { throw null; }
+        public virtual System.Threading.Tasks.Task<string> ReadToEndAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
```

Comment:
There are a few other TextReader-derived types in dotnet/runtime, e.g. a couple in System.Console, and NullTextReader in corelib.  We should likely override the new overloads on those as well. (I'm not sure why NullTextReader doesn't already override most of the virtuals, but it seems it should, for perf.)  Same goes for NullStreamReader.

Reply:
> I'm not sure why NullTextReader doesn't already override most of the virtuals, but it seems it should, for perf.

Is it OK to add all the overrides as part of this PR, or should that be separate work (and this PR just add `ReadXAsync`)?

Reply:
Separate is fine.

Reply:
I have a commit at https://github.com/bgrainger/runtime/commit/92e92cca34282fbb9cb25e8a80a46187d8f4d6c9 ready to go once this PR is merged.